src/lv_widgets/lv_textarea.c

```diff
@@ -244,7 +244,7 @@ void lv_textarea_add_char(lv_obj_t * ta, uint32_t c)
 #endif
 
     ta_insert_replace = NULL;
-    lv_event_send(ta, LV_EVENT_INSERT, letter_buf);
+    
     if(ta_insert_replace) {
```

Comment:
This `if` still belongs to the event sending so it should with it.

`ta_insert_replace` is will be set to the new data to insert.

Reply:
I didn't quite understand what you meant... am I supposed to put it inside the `if` statement? The idea is that it shouldn't be sent if `lv_textarea_add_text` is going to be called recursively anyways (line 253).

Reply:
I see but this block(s) belongs together. Probably it'd be better to add these blocks to a function.
```c
    ta_insert_replace = NULL;
    lv_event_send(ta, LV_EVENT_INSERT, txt);
    if(ta_insert_replace) {      /*See if `ta_insert_replace` was set in `lv_textarea_set_insert_replace`*/
        if(ta_insert_replace[0] == '\0') return; /*Drop this text*/

        /*Add the replaced text directly it's different from the original*/
        if(strcmp(ta_insert_replace, txt)) {
            lv_textarea_add_text(ta, ta_insert_replace);
            return;
        }
    }
```

Reply:
Do we want `LV_EVENT_INSERT` sent a second time in that case (with the replaced text)?

Reply:
Hmm, it seems we talk at cross purposes. I added a commit to show what I mean. I placed the event where you suggested.

Reply:
Hi @embeddedt & @kisvegabor,

I have tested this and the ```LV_EVENT_INSERT``` no longer is firing twice so I think the patch is good for merging.

Kind Regards,

Pete

Reply:
Okay; I think what @kisvegabor did will work, so if he approves we can merge this.